analyses/malawi/scripts/mwi_chirps_dry_spell_viz.R

```diff
@@ -13,118 +14,174 @@ source("scripts/mwi_chirps_dry_spell_viz_hm_functions.R")
-df_dry_spells <- read.csv(paste0(dry_spell_dir, 'dry_spells_during_rainy_season_list_2000_2020_adm3.csv'))
-df_rainy_season <- read.csv(paste0(dry_spell_dir, 'rainy_seasons_detail_2000_2020_ADM3.csv'))
-plot_title=glue("20 years of dry spells in Malawi's admin 3 regions, 2000-2020")
-output_path_hm=paste0(exploration_dry_spell_dir, glue('mwi_viz_hm_dry_spell_mean_adm3.png'))
-hm_adm3 <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,output_path_hm)
-# ggsave(output_path_hm,plot = hm_adm3, width=20,height=15)
-
-## ADMIN1 pixel-based
-#define file paths
-threshold <- 50
-df_dry_spells <- read.csv(paste0(exploration_dry_spell_dir,glue('dryspells_pixel_adm1_th{threshold}_viz.csv')))
-df_rainy_season <- read.csv(paste0(exploration_dry_spell_dir, "rainy_seasons_detail_2000_2020_per_pixel_adm1.csv"))
-
-#Set variables for heatmap
-y_label="Admin 1 region"
-plot_title=glue('Observed dry spells by CHIRPS pixel-based aggregation on ADM1, {threshold}% of pixels threshold')
-output_path_hm=paste0(exploration_dry_spell_dir, glue('viz_chirps_pixel_adm1_th_{threshold}_dryspell_hm_test.png'))
-ds_flatdata=TRUE
-# #TODO: add option to add yticks labels
-# yticks_text=TRUE
-hm_adm1 <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,output_path_hm,ds_flatdata,yticks_text)
-# ggsave(output_path_hm,plot = hm_adm1, width=20,height=15)
-
-#TODO: create plots for the less than xmm per day methodology
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/mean_back_dry_spell_hm.png'))
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/dry_spell_hm_consecutive_days_4mm.png'), width = 7.55, height = 7.82, units = "in", dpi = 300)
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/dry_spell_hm_consecutive_days_2mm.png'), width = 7.55, height = 7.82, units = "in", dpi = 300)
-
-
-
-
-
-### Make heatmap of the match of observed dry spells with different aggregation methodologies and different indicators
-## Mean vs >=50% pixels
-#Set general variables for heatmap
-dry_spell_match_values=c(10, 0, 13,12,11)
-match_values_labels=c("Rainy season", "Dry season", glue('mean and >={threshold} dry spell'),'mean dry spell',glue('>={threshold} dry spell'))
-color_scale=c('#b3e7ff', '#fff2d6','#0063B3' ,'#F2645A',"#78D9D1")
-y_label="Admin 2 region"
+ ## mean
+ df_dry_spells <- read.csv(paste0(dry_spell_dir,'dry_spells_during_rainy_season_list_2000_2020_mean_back.csv'))
+ df_rainy_season <- read.csv(paste0(dry_spell_dir, "rainy_seasons_detail_2000_2020_mean_back.csv"))
+ plot_title=glue("20 years of dry spells in Malawi's admin 2 regions, 2000-2020") #not indicating aggregation method as this is our standard
+ output_path_hm=paste0(exploration_dry_spell_dir, glue('mwi_viz_hm_dry_spell_mean_adm2.png'))
+ hm_mean <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,output_path_hm)
```

Comment:
Hmm here I'm getting ```Error in if (ds_flatdata) { : argument is not interpretable as logical```

Reply:
Might be missing the right ```ds_flatdata``` arg here

Reply:
Oops sorry for that mess.. The output path was still being given as argument, while that was removed from the function